src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,305 @@
+
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.File;
+import java.io.FileReader;
+import java.io.BufferedReader;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.util.Scanner;
+
+import java.util.ArrayList;
+
+public class Duke {
+
+    public static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void main(String[] args) {
+
+        openFile("./data/duke.txt");
+        printStartMessage();
+
+        String input;
+        Scanner in = new Scanner(System.in);
+
+        do {
+            input = in.nextLine();
+            try {
+                processCommand(input);
+            } catch (CommandNotRecognisedException e) {
+                System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+                printDivider();
+            }
+        } while (!input.equals(Command.COMMAND_BYE));
+
+        File fileName = new File("./data/duke.txt");
+        writeToFile(fileName);
+    }
+
+    public static void openFile(String input) {
+        File file = new File(input);
+        readFile(file);
+    }
+
+    public static void readFile(File file) {
+        try {
+            FileReader fr = new FileReader(file);
+            BufferedReader br = new BufferedReader(fr);
+
+            String line;
+            while((line=br.readLine())!=null) {
+                initialiseTaskList(line);
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("You have no pre-existing tasks :)");
+        }  catch (IOException e) {
+            System.out.println("Error reading file :(");
+        }
+    }
+
+    public static void writeToFile(File filePath) {
+        try {
+            PrintWriter pw = new PrintWriter(filePath);
+            writeTaskToFile(pw);
+
+        }  catch (FileNotFoundException e) {
+            File dir = new File("./data");
+            if (!dir.isDirectory()) {
+                System.out.println("Data directory not found. Creating new data directory");
+                Path path = Paths.get("./data");
+                try {
+                    Files.createDirectories(path);
+                } catch (IOException f) {
+                    System.out.println("Failed to create directory");
+                }
+            }
+            if (!filePath.isFile()) {
+                System.out.println("File not found. Creating new text file");
+                Path fileLoc = Paths.get("./data/duke.txt");
+                try {
+                    Files.createFile(fileLoc);
+                } catch (IOException f) {
+                    System.out.println("Failed to create file");
+                }
+            }
+
+        }
+    }
+
+    public static void writeTaskToFile(PrintWriter pw) {
+        for (int i = 0; i < tasks.size(); ++i) {
+            pw.println(tasks.get(i).printToFile());
+        }
+        pw.close();
+    }

```

Comment:
Since this part of code is specifically used for Storage, should we create a Storage class instead of put it in the main class?